qiita_db/base.py

```diff
@@ -214,6 +212,10 @@ def __ne__(self, other):
         r"""Self and other are not equal based on type and database id"""
         return not self.__eq__(other)
 
+    def __hash__(self):
```

Comment:
The hash is useful for doing sets and/or dictionaries. In the following example:
```python
s1 = Study(1)
s2 = Study(1)
s = set(s1, s2)
len(s)
```
w/o the hash function `len(s) = 2`. However, with this hash function `len(s) = 1` which is correct given that s1 and s2 represents the same study. This function was not needed until now because we where moving ids around, but since we are using the objects now, we need to hash correctly these objects.